test_conformance/profiling/execute_multipass.cpp

```diff
@@ -161,6 +139,27 @@ static int run_kernel( cl_device_id device, cl_context context, cl_command_queue
         return -1;
     }
 
+    err = clGetDeviceInfo(device, CL_DEVICE_MAX_WORK_ITEM_SIZES,
+                          3 * sizeof(size_t), (size_t *)localThreads, NULL);
+    if (err)
+    {
+        log_error("clGetDeviceInfo(CL_DEVICE_MAX_WORK_ITEM_SIZES) failed\n");
+        return -1;
+    }
+    err = clGetKernelWorkGroupInfo(kernel[0], device, CL_KERNEL_WORK_GROUP_SIZE,
+                                   sizeof(size_t), &maxWorkgroupSize, NULL);
+    if (err)
+    {
+        log_error("clGetDeviceInfo(CL_KERNEL_WORK_GROUP_SIZE) failed\n");
+        return -1;
+    }
+    localThreads[0] =
+        std::min({ localThreads[0], threads[0], maxWorkgroupSize });
+    localThreads[1] = std::min(
+        { localThreads[1], threads[1], maxWorkgroupSize / localThreads[0] });
+    localThreads[2] =
+        std::min({ localThreads[2], threads[2],
+                   maxWorkgroupSize / (localThreads[0] * localThreads[1]) });
```

Comment:
This is fine as-is, but if we end up doing this in a lot of places we probably ought to add a helper function in the harness for it.

Note: I did find `get_max_common_3D_work_group_size` and `get_max_allowed_work_group_size`, but neither appear to be a replacement for this code.